packages/ui-app/src/InputNumber.tsx

```diff
@@ -0,0 +1,151 @@
+    const { error, info, warn } = this.state;
+
+    return (
+      <div
+        className={classes('ui--InputNumber', className)}
+        style={style}
+      >
+        <Input
+          className={className}
+          defaultValue={defaultValue || '0'}
+          error={error}
+          info={info}
+          isError={isError}
+          label={label}
+          maxLength={maxLength}
+          onChange={this.onChange}
+          onKeyDown={this.onKeyDown}
+          onKeyUp={this.onKeyUp}
+          placeholder={placeholder}
+          style={style}
+          type='text'
+          warn={warn}
+          withLabel={withLabel}
+        />
+      </div>
+    );
+  }
+
+  onChange = (value: string): void => {
+    const { onChange, t } = this.props;
+
+    try {
+      const { isValid, errorMessage, infoMessage, warnMessage } = isValidBalance(value, t);
+
+      this.setState({
+        error: !isValid && errorMessage ? errorMessage : '',
+        info: isValid && infoMessage ? infoMessage : '',
+        warn: isValid && warnMessage ? warnMessage : ''
+      });
+
+      let valueBN: BN;
+      valueBN = new BN(value || '0');
+
+      if (!onChange) {
+        return;
+      }
+
+      onChange({
+        isValid,
+        value: valueBN
+      });
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  onKeyDown = (event: React.KeyboardEvent<Element>): void => {
+    const { isPreKeyDown } = this.state;
+    const eventValue = (event.target as HTMLInputElement).value;
+
+    // only allow user balance input to contain one instance of '.' for decimals.
+    // prevent use of shift key
+    if (
+      (keydown.isDuplicateDecimalPoint(event.key, eventValue)) ||
+      (keydown.isShift(event.shiftKey))
+    ) {
+      event.preventDefault();
+      return;
+    }
+
+    if (KEYS_ALLOWED.includes(event.key)) {
+      return;
+    }
+
+    if (KEYS_PRE.includes(event.key)) {
+      this.setState({ isPreKeyDown: true });
+    }
+
+    // allow users to to use cut/copy/paste combinations, but not non-numeric letters individually
+    if (
+      (keydown.isSelectAll(event.key, isPreKeyDown)) ||
+      (keydown.isCut(event.key, isPreKeyDown)) ||
+      (keydown.isCopy(event.key, isPreKeyDown)) ||
+      (keydown.isPaste(event.key, isPreKeyDown))
+    ) {
+      return;
+    }
+
+    // prevent input of non-integer values (allow numeric including from keyboards with numpad)
+    if (keydown.isNonNumeric(event.key)) {
+      console.error('Balance must be a positive number');
+      event.preventDefault();
+      return;
+    }
+
+    const inputBN = new BN(eventValue);
+    const maxSafeIntegerBN = new BN(Number.MAX_SAFE_INTEGER);
+
+    if (inputBN.gt(maxSafeIntegerBN)) {
+      event.preventDefault();
+      return;
+    }
+  }
+
+  onKeyUp = (key: string): void => {
+    if (KEYS_PRE.includes(key)) {
+      this.setState({ isPreKeyDown: false });
+    }
+  }
+}
+
+export default InputNumber;
```

Comment:
I see the testing comment, however when we specify `t` the component needs to be wrapped in a translate. We cannot have components using this pass through `t`. So it _needs_ to be wrapped. (The testing setup is incorrect if it is not working, i.e. i18next is not being initialised)